### PR TITLE
feat(core): Improve debugging of sub-workflows

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary

Saves the Execution ID of sub and parent workflows and so makes it now easily possible to jump to the corresponding sub or parent workflow.

based on https://github.com/n8n-io/n8n/pull/8608

Screenshot 2024-11-12 at 12 52 52

Screenshot 2024-11-12 at 12 53 31

Screenshot 2024-11-12 at 12 53 17

Screenshot 2024-11-12 at 13 02 01

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/workflow-execute-additional-data.ts [hunk 1]
```diff
@@ -776,7 +776,41 @@ export async function executeWorkflow(
...
+		executionId,
+		runData,
+		workflowData,
+	);
+
+	if (options.doNotWaitToFinish) {
+		return { executionId, data: [null] };
+	}
+
+	return await executionPromise;
```
Comment (mutdmour) [inline packages/cli/src/workflow-execute-additional-data.ts:804]: here to allow to skip waiting for subexecutions, for example in execute workflow now
Screenshot 2024-11-12 at 13 00 00

**Review by RicardoE105 — DISMISSED** [review]
Tested it locally and everything seems to be working as expected. Awesome work 🙌

File: packages/cli/package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "start:default": "cd bin && ./n8n",
     "start:windows": "cd bin && n8n",
     "test": "pnpm test:sqlite",
+    "test:dev": "N8N_LOG_LEVEL=silent DB_TYPE=sqlite jest --watch",
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/cli/package.json:28]: Yes, it's very helpful for running tests

**Review by RicardoE105 — APPROVED** [review]

**Review by RicardoE105 — DISMISSED** [review]

**Review by RicardoE105 — APPROVED** [review]

**Review by RicardoE105 — DISMISSED** [review]

**Review by RicardoE105 — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.69.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.69.0)

**Pull request merged**